Proposal for Generic Mapping Tools governance structure

maxrjones opened the pull request:
Here is a proposal for a governance structure for GMT, based on [Pangeo's model](https://github.com/pangeo-data/governance). I have also considered the Jupyter docs (xref #6), however think a simpler approach would fit better here.

Please leave comments and suggestions on this framework via GitHub between now and May 10. We will vote on this document and any steering committee updates between May 10 and 17, 2024

weiji14 reviewed (commented):
Thanks @maxrjones for initiating this governance doc! Looks really good overall already, but just opening the conversation on a few specific points.

seisman commented on governance.md:
```diff
@@ -0,0 +1,183 @@
+Outcomes and decisions should be clearly communicated to the Community. This can
+take three different forms:
+
+* New challenges for The Project will be detailed as a GitHub issue for the Community
+  to discuss.
+* Resolved technical deadlocks will result in an explanation on the relevant
+  issue and/or closure of the relevant pull request.
+* Other decisions will be detailed as a narrative blog post.
```
> 1. Do we agree on using the GitHub as the medium of discussion, or should we also consider the GMT forum for certain non-code-related issues?

Maybe it's better to discuss in the GitHub organization Discussions (not enabled yet).

> 3. We don't really have a blog for GMT anywhere (as far as I'm aware). Maybe state that the decision will be published somewhere on the website at https://www.generic-mapping-tools.org or on the GMT forum?

It's time to have a GMT blog!

maxrjones replied:
> > Do we agree on using the GitHub as the medium of discussion, or should we also consider the GMT forum for certain non-code-related issues?
> Maybe it's better to discuss in the GitHub organization Discussions (not enabled yet).

I enabled discussions on this repository. We could instead enable organization discussions, but would need to decide which repository would host the discussions. I also added that discussions could happen on the Forum.

> Closing of an issue/PR is usually done at the discretion of a repository maintainer, though I guess we could involve the steering committee in certain situations.

I didn't change anything here. I'm open to removing it, but think it could be helpful to keep even if it is rarely needed.

> We don't really have a blog for GMT anywhere (as far as I'm aware). Maybe state that the decision will be published somewhere on the website at [generic-mapping-tools.org](https://www.generic-mapping-tools.org/) or on the GMT forum?

I added a note that it could be an announcement on the website. I agree that starting a blog would be nice, so I left that point.

joa-quim commented:
Hi Max,

Thanks for this proposal that reads very nicely, but I'm afraid that in my view it is a bit too extended (lack a better word right now) regarding the GMT reality. For example, I would drop the _Conflict of Interest _ section all together.

Regarding the Committee role, I find it a bit lengthy and I'm not sure I understand this point:

> Make decisions about specific technical issues, features, bugs and pull requests. They are the primary mechanism of guiding the code review process and merging pull requests.

We certainly do not want that PRs will need the approval by the Committee to be merged.

I took a look at the [GDAL Committee Guidelines]( https://gdal.org/development/rfc/rfc1_pmc.html) and I like very much the simplicity of the “When is Vote Required?” so I made some changes to it and propose that we adopt it as

```
Admission of new members (under proposal of a committee member)

Granting repository commit access to new contributors (under proposal of a committee member). 

Removal of repository commit access (same procedure as for granting).

Anything that could cause backward compatibility issues.

Adding substantial amounts of new code.

When releases should take place.

Anything that might be controversial.
```

Note that is cover the very important and not covered point of granting commit access to new contributors.

Finally, regarding the starting Committee composition, I fully agree on having elements from MB-System and GMTSAR. Dave Caress and Eric Xu are perfect choices. The other members should be people more involved in current and past GMT development, so I propose:

- Joaquim Luis
- Remko Scharroo
- Dongdong Tian
- Federico Esteban
- Eric Xu
- Dave Caress

Of course, if any other person (namely someone from EarthScope) feel that he/she would like to make part of the committee we are open to discussion.

maxrjones commented on governance.md:
```diff
@@ -0,0 +1,183 @@
+  individuals.
+* Make decisions about specific technical issues, features, bugs and pull
+  requests. They are the primary mechanism of guiding the code review process and
+  merging pull requests.
```
Based on yours and Joaquim's feedback, I removed this bullet completely in https://github.com/GenericMappingTools/oversight/pull/7/commits/8d151094715e5871997498fcbcdd8c42d0b28182

maxrjones commented on governance.md:
```diff
@@ -0,0 +1,183 @@
+
+To become eligible for being a Steering Committee Member an individual must be a
+Project Contributor who has produced contributions that are substantial in
+quality and quantity, and sustained over at least six months.
```
Yes, definitely. Good point, I clarified this in https://github.com/GenericMappingTools/oversight/pull/7/commits/ae221c9ec34ba54fddba4f0ec51dd483617222b0

maxrjones commented:
@weiji14, @seisman, @joa-quim thank you for the feedback.

Here are some changes based on this feedback, in addition to the in-line responses above:

> For example, I would drop the _Conflict of Interest _ section all together.

I removed this section

> Regarding the Committee role, I find it a bit lengthy and I'm not sure I understand this point:
> 
> > Make decisions about specific technical issues, features, bugs and pull requests. They are the primary mechanism of guiding the code review process and merging pull requests.
> 
> We certainly do not want that PRs will need the approval by the Committee to be merged.

I removed this bullet

> I took a look at the [GDAL Committee Guidelines](https://gdal.org/development/rfc/rfc1_pmc.html) and I like very much the simplicity of the “When is Vote Required?” so I made some changes to it and propose that we adopt it as
> 
> Admission of new members (under proposal of a committee member)
> 
> Granting repository commit access to new contributors (under proposal of a committee member). 
> 
> Removal of repository commit access (same procedure as for granting).
> 
> Anything that could cause backward compatibility issues.
> 
> Adding substantial amounts of new code.
> 
> When releases should take place.
> 
> Anything that might be controversial.
> Note that is cover the very important and not covered point of granting commit access to new contributors.

To me this seems similar to the point about about not wanting PRs to need approval by the Committee. I think that granting commit permissions should be at the discretion of the specific repository maintainers and not the Steering Committee. The steering committee should be providing a higher level of guidance. We could add that the Steering Committee can resolve conflicts in whether someone is granted commit access.


> Finally, regarding the starting Committee composition, I fully agree on having elements from MB-System and GMTSAR. 

> Dave Caress and Eric Xu are perfect choices. The other members should be people more involved in current and past GMT development, so I propose:
> 
> Joaquim Luis
> Remko Scharroo
> Dongdong Tian
> Federico Esteban
> Eric Xu
> Dave Caress
> Of course, if any other person (namely someone from EarthScope) feel that he/she would like to make part of the committee we are open to discussion.

This is the next topic of discussion after we get this merged. I suggest we discuss separately in the Discussions section in this repository.

Can you please all give this one more review and then I will email the broader group to see if there are any objections to adopting these governance rules?

seisman commented on governance.md:
```diff
@@ -0,0 +1,162 @@
+
+#### Committee Membership
+
+To become eligible for being a Steering Committee Member an individual must be a
+Project Contributor who has produced contributions that are substantial in
+quality and quantity, and sustained over at least six months. Project contributions
```
Perhaps we should also mention that members must have a GitHub account to participant in discussions on GitHub.

maxrjones replied:
Done!

seisman commented on governance.md:
```diff
@@ -0,0 +1,162 @@
+#### Expectations for Committee Members
+
+Steering Committee members are expected to commit to the following activities:
+- Attend a one-hour steering committee meeting every six months.
```
Given the global distribution of the Committee members, not all of them will be able to attend the meeting due to the time zones involved.

seisman reviewed (commented):
Looks good to me.

joa-quim commented:
> I think that granting commit permissions should be at the discretion of the specific repository maintainers and not the Steering Committee. The steering committee should be providing a higher level of guidance. We could add that the Steering Committee can resolve conflicts in whether someone is granted commit access.

I think this is the main point. For me it makes no sense to have a steering committee that is not constituted in large majority of people directly involved in the GMT development/maintenance. Sorry but "_The steering committee should be providing a higher level of guidance_" is something that doesn't make much sense to me and that I don't believe will ever occur. It makes me think on @WalterHFSmith stories when he described GEBCO meetings where Admirals wanted to steer the work of volunteers, which obviously didn't want to care about that steering.

For me, agreeing on the steering committee constitution is one the basic points of the governance plan. And I think that Dondong's comment about the _global distribution of the Committee members_ also reflects that he is anticipating the committee constitution.

maxrjones commented:
> > I think that granting commit permissions should be at the discretion of the specific repository maintainers and not the Steering Committee. The steering committee should be providing a higher level of guidance. We could add that the Steering Committee can resolve conflicts in whether someone is granted commit access.
> 
> I think this is the main point. For me it makes no sense to have a steering committee that is not constituted in large majority of people directly involved in the GMT development/maintenance. Sorry but "_The steering committee should be providing a higher level of guidance_" is something that doesn't make much sense to me and that I don't believe will ever occur. It makes me think on @WalterHFSmith stories when he described GEBCO meetings where Admirals wanted to steer the work of volunteers, which obviously didn't want to care about that steering.
> 
> For me, agreeing on the steering committee constitution is one the basic points of the governance plan. And I think that Dondong's comment about the _global distribution of the Committee members_ also reflects that he is anticipating the committee constitution.

@joa-quim could you please provide some in-line suggestions via a code review for what you'd like to see in the governance structure documentation? For example, you could add a line that states "the majority of people in the steering committee should be directly involvement in the GMT [or wrappers] development/maintenance". This would be more expedient than me trying to revise based on your comments and pinging again for review. Thanks!

maxrjones commented on governance.md:
```diff
@@ -0,0 +1,162 @@
+are hosted under the generic-mapping-tools.org domain. Examples of Project Services
+include the Generic Mapping Tools website (https://www.generic-mapping-tools.org/),
+the Generic Mapping Tools data servers (listed at https://www.generic-mapping-tools.org/mirrors/),
+and the Generic Mapping Tools Discourse Forum (https://forum.generic-mapping-tools.org/).
```
```suggestion the Generic Mapping Tools Discourse Forum (https://forum.generic-mapping-tools.org/), and
the PyGMT project domain (https://www.pygmt.org).
```

maxrjones replied:
```suggestion
the Generic Mapping Tools Discourse Forum (https://forum.generic-mapping-tools.org/), and the
PyGMT project domain (https://www.pygmt.org).
```

maxrjones commented on governance.md:
```diff
@@ -0,0 +1,162 @@
+LGPL and BSD, developed openly and  hosted in public GitHub repositories under the 
+GenericMappingTools GitHub organization. Examples of Project Software include the 
+GMT C library and the Python (PyGMT), Julia (GMT.jl), and MATLAB (GMT/MEX) wrappers.
+The Services run by The Project consist of public websites and web-services that
```
good idea, added below

maxrjones commented on governance.md:
```diff
@@ -0,0 +1,163 @@
+
+To become eligible for being a Steering Committee Member an individual must be a
+Project Contributor who has produced contributions that are substantial in
+quality and quantity, and sustained over at least six months. Project contributions
```
```suggestion
quality and quantity, and sustained over at least six months. Steering Committee Members should
have a GitHub account to participate in discussions and document decisions. Project contributions
```

maxrjones commented on steering-committee-membership.md:
```diff
@@ -0,0 +1,9 @@
+# Steering Committee Membership
+
+The Generic Mapping Tools Steering Committee members are:
+
+- David Sandwell (Scripps) - Chair
+- Dave Caress (MBARI)
+- Steve Diggs (Scripps)
+- Dan Bassett (GNS Science, New Zealand)
+- Khalid Soofi (ConocoPhillips)
```
With the creation of a formal governance structure, the Generic Mapping Tools Steering
Committee with be re-constituted with membership decided by active project developers
and previous core developers in June 2024.

maxrjones commented on steering-committee-membership.md:
```diff
@@ -0,0 +1,9 @@
+# Steering Committee Membership
+
+The Generic Mapping Tools Steering Committee members are:
+
+- David Sandwell (Scripps) - Chair
+- Dave Caress (MBARI)
+- Steve Diggs (Scripps)
+- Dan Bassett (GNS Science, New Zealand)
+- Khalid Soofi (ConocoPhillips)
```
```suggestion
With the creation of a formal governance structure, the Generic Mapping Tools Steering
Committee with be re-constituted with membership decided by active project developers
and previous core developers in June 2024.
```

maxrjones replied:
FYI I suggested this change so that we can re-establish the Steering Committee members based on this governance structure in a separate discussion.

Please thumbs up if you agree or comment if you disagree.

joa-quim replied:
I'm fine with this solution to move forward.

maxrjones commented:
FYI I suggested this change so that we can re-establish the Steering Committee members based on this governance structure in a separate discussion.

https://github.com/GenericMappingTools/oversight/pull/7/files#r1620969400

Please thumbs up if you agree or comment if you disagree.

maxrjones commented:
Alright, I think it's time to move this forward 🎉 

Please approve this pull request if you are alright with this structure. I'll merge it on Sunday at 8 pm EST if there are a few approvals and no new comments. Thanks everyone!

WalterHFSmith reviewed (approved):
Thank you, everyone, for good work on this.

seisman commented:
@maxrjones Any progress on voting Steering Committee membership?

maxrjones commented:
> @maxrjones Any progress on voting Steering Committee membership?

Sorry for the delay - please see #9
